core/src/main/java/org/hisp/dhis/android/sdk/synchronization/domain/enrollment/EnrollmentSynchronizer.java

```diff
@@ -48,6 +53,20 @@ public void sync(Enrollment enrollment) {
         }
     }
 
+    public void sync(List<Enrollment> enrollments) {
+        Collections.sort(enrollments, new Enrollment.EnrollmentComparator());
+
+        for (Enrollment enrollment : enrollments) {
+            if(enrollment.isFromServer()){
+                continue;
+            }
+            if(enrollment.getCreated()==null && (enrollment.getStatus().equals(Enrollment.CANCELLED) || enrollment.getStatus().equals(Enrollment.COMPLETED))) {
```

Comment:
I think that We can refactor this code for to do more readable, for example: Only one invocation to sync method and inside to sync method:
1 - Sync enrollment
2 - Sync events
3 - if enrollment is cancelled o completed then sync enrollment again